File: OrangeJuice.Server.Core/Configuration/ConfigurationConnectionStringProvider.cs
namespace OrangeJuice.Server.Configuration
{
	public class ConfigurationConnectionStringProvider : IConnectionStringProvider
	{
		private readonly IConfigurationProvider _configurationProvider;

		public ConfigurationConnectionStringProvider(IConfigurationProvider configurationProvider)
		{
			_configurationProvider = configurationProvider;
		}

		public string GetDefaultConnectionString()
		{
			return _configurationProvider.GetValue("sql:ConnectionString");
		}
	}
}

File: OrangeJuice.Server.Core.FSharp/Configuration/ConfigurationConnectionStringProvider.fs
namespace OrangeJuice.Server.FSharp.Configuration

open OrangeJuice.Server.Configuration

type ConfigurationConnectionStringProvider(configurationProvider : IConfigurationProvider) =
    interface IConnectionStringProvider with
        member this.GetDefaultConnectionString() : string =
            configurationProvider.GetValue("sql:ConnectionString")